src/OpenTelemetry.AutoInstrumentation/Instrumentations/Kafka/Integrations/ProducerProduceSyncIntegration.cs

```diff
@@ -0,0 +1,135 @@
+using OpenTelemetry.Context.Propagation;
+// ReSharper disable ConditionIsAlwaysTrueOrFalseAccordingToNullableAPIContract
+
+namespace OpenTelemetry.AutoInstrumentation.Instrumentations.Kafka.Integrations;
+
+/// <summary>
+/// Kafka sync produce instrumentation
+/// </summary>
+[InstrumentMethod(
+    assemblyName: IntegrationConstants.ConfluentKafkaAssemblyName,
+    typeName: IntegrationConstants.ProducerTypeName,
+    methodName: IntegrationConstants.ProduceSyncMethodName,
+    returnTypeName: ClrNames.Void,
+    parameterTypeNames: new[] { IntegrationConstants.TopicPartitionTypeName, IntegrationConstants.MessageTypeName, IntegrationConstants.ActionOfDeliveryReportTypeName },
+    minimumVersion: IntegrationConstants.MinVersion,
+    maximumVersion: IntegrationConstants.MaxVersion,
+    integrationName: IntegrationConstants.IntegrationName,
+    type: InstrumentationType.Trace)]
+public static class ProducerProduceSyncIntegration
+{
+    internal static CallTargetState OnMethodBegin<TTarget, TTopicPartition, TMessage, TDeliveryHandler>(
+        TTarget instance, TTopicPartition topicPartition, TMessage message, TDeliveryHandler deliveryHandler)
+        where TMessage : IKafkaMessage, IDuckType
+    {
+        // duck type created for message is a struct
+        if (message.Instance is null || topicPartition is null || !topicPartition.TryDuckCast<ITopicPartition>(out var duckTypedTopicPartition))
+        {
+            // invalid parameters, exit early
+            return CallTargetState.GetDefault();
+        }
+
+        string? spanName = null;
+        if (!string.IsNullOrEmpty(duckTypedTopicPartition.Topic))
+        {
+            spanName = $"{duckTypedTopicPartition.Topic} {MessagingAttributes.Values.PublishOperationName}";
+        }
+
+        spanName ??= MessagingAttributes.Values.PublishOperationName;
+        var activity = KafkaCommon.Source.StartActivity(name: spanName, ActivityKind.Producer);
+        if (activity is not null)
+        {
+            message.Headers ??= MessageHeadersHelper<TTopicPartition>.Create();
+            Propagators.DefaultTextMapPropagator.Inject<IKafkaMessage>(
+                new PropagationContext(activity.Context, Baggage.Current),
+                message,
+                KafkaCommon.MessageHeaderValueSetter);
+
+            if (activity.IsAllDataRequested)
+            {
+                KafkaCommon.SetCommonAttributes(
+                    activity,
+                    MessagingAttributes.Values.PublishOperationName,
+                    duckTypedTopicPartition.Topic,
+                    duckTypedTopicPartition.Partition,
+                    message.Key,
+                    instance.DuckCast<IClientName>());
+
+                activity.SetTag(MessagingAttributes.Keys.Kafka.IsTombstone, message.Value is null);
+            }
+
+            // Store as state information if delivery handler was set
+            return new CallTargetState(activity, deliveryHandler is null);
```

Comment:
By passing a `bool` as the `object state` argument, you're unnecessarily doing a box operation. I suggest you pass the `deliveryHandler` object as-is and later do the null-check

Reply:
fixed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3055/commits/fb2e35bf6a6b09871bec0195dfb2943655521699